node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,339 @@
+                                true,
+                            ),
+                        ).rejects.toThrow(RequestError);
+                        await client.updateConnectionPassword(
+                            NEW_PASSWORD,
+                            true,
+                        );
+                        await client.configSet({
+                            requirepass: "",
+                        });
+                    } finally {
+                        client?.close();
+                    }
+                }, protocol);
+            });
+
+            /**
+             * Test replacing connection password with immediate re-authentication.
+             */
+            it(
+                "test_update_connection_password_with_immediateAuth",
+                async () => {
+                    await runTest(async (client: BaseClient) => {
+                        // Set server password
+                        await client.configSet({ requirepass: NEW_PASSWORD });
+
+                        // Update client password with re-auth
+                        await expect(
+                            client.updateConnectionPassword(NEW_PASSWORD, true),
+                        ).resolves.toBe("OK");
+
+                        // Verify client works with new auth
+                        await client.set("test_key", "test_value");
+                        const value = await client.get("test_key");
+                        expect(value).toEqual("test_value");
+
+                        // Clean up
+                        await client.configSet({ requirepass: "" });
+                    }, protocol);
+                },
+                TIMEOUT,
+            );
+
+            /**
+             * Test changing server password when connection is lost before password update.
+             * Verifies that the client will not be able to reach the inner core and return an error.
+             */
+            it("test_update_connection_password_connection_lost_before_password_update", async () => {
+                await runTest(async (client: BaseClient) => {
+                    if (client instanceof GlideClusterClient) {
+                        return;
+                    }
+
+                    // Set a key to ensure connection is established
+                    await client.set("test_key", "test_value");
+                    // Update server password
+                    await client.configSet({ requirepass: NEW_PASSWORD });
+                    // Kill client connections
+                    await managementClient.customCommand([
+                        "CLIENT",
+                        "KILL",
+                        "TYPE",
+                        "normal",
+                    ]);
+                    // Try updating client password without immediate re-auth
+                    await expect(
+                        client.updateConnectionPassword(NEW_PASSWORD, false),
```

Comment:
if this passes `true` does this command work?  
should we test the `true` case too?

Reply:
I'll add also. But it will fail. There is no one listening on the other side to send the command.